product/cmd/grpc/client/client_test.go

```diff
@@ -0,0 +1,131 @@
+package main
```

Comment:
Re the whole file: nice!

You pick these things up really quickly. I think my only piece of remaining advice would be to make sure that you appreciate why and don't just do things simply because I suggest them. In this case it's to avoid circular arguments in testing because ExpectCall does nothing more than mirror your implementation.

Reply:
I understand why I had to restructure the test, I had similar thoughts about testing the actual implementation but didn't have a better idea at the time.